### PR TITLE
[ISSUE #3628] set naming client updateTask interval more flexible

### PR DESCRIPTION
**Pull request by horizonzy**
## What is the purpose of the change

fix-#3628

### CODE REVIEW

File: api/src/main/java/com/alibaba/nacos/api/naming/pojo/ServiceInfo.java [hunk 1]
```diff
@@ -48,6 +48,8 @@
     
     private long cacheMillis = 1000L;
     
+    private long failCount = 0L;
```
Comment (KomachiSion) [inline api/src/main/java/com/alibaba/nacos/api/naming/pojo/ServiceInfo.java:51]: fail count should not be attribute of ServiceInfo. It should be maintained in UpdateTask.

File: client/src/main/java/com/alibaba/nacos/client/naming/core/HostReactor.java [hunk 2]
```diff
@@ -411,7 +412,7 @@ public void run() {
                     return;
                 }
                 
-                delayTime = serviceObj.getCacheMillis();
+                delayTime = serviceObj.getCacheMillis() + Math.min(serviceObj.getFailCount() * 1000L, 10000L);
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/naming/core/HostReactor.java:415]: The exception delay time should be calculate in `catch`
And the algorithm should be enhance. Such as double time then last time. And the max delay time is not more than 60s.

File: naming/src/main/java/com/alibaba/nacos/naming/controllers/InstanceController.java [hunk 3]
```diff
@@ -516,21 +516,21 @@ public ObjectNode doSrvIpxt(String namespaceId, String serviceName, String agent
         ClientInfo clientInfo = new ClientInfo(agent);
         ObjectNode result = JacksonUtils.createEmptyJsonNode();
         Service service = serviceManager.getService(namespaceId, serviceName);
-        
+        long cacheMillis = switchDomain.getDefaultCacheMillis();
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/controllers/InstanceController.java:519]: If updPort > 0 , it should be pushCacheMillis. 
See the logic in 541 line.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: client/src/main/java/com/alibaba/nacos/client/naming/core/HostReactor.java [hunk 4]
```diff
@@ -412,12 +419,14 @@ public void run() {
...
                 delayTime = serviceObj.getCacheMillis();
-                
+                failCount = 0;
             } catch (Throwable e) {
+                failCount++;
                 NAMING_LOGGER.warn("[NA] failed to update serviceName: " + serviceName, e);
             } finally {
                 if (delayTime > 0) {
-                    executor.schedule(this, delayTime, TimeUnit.MILLISECONDS);
+                    executor.schedule(this, Math.min(delayTime << failCount, DEFAULT_DELAY * 60),
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/naming/core/HostReactor.java:428]: The implementation may cause `delayTime << failCount` become negative number when failCount=54.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: client/src/main/java/com/alibaba/nacos/client/naming/core/HostReactor.java (hunk 4 above)
Comment (horizonzy) [reply client/src/main/java/com/alibaba/nacos/client/naming/core/HostReactor.java]: now, the failCunt cat't limit 6.

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**